launcher/modplatform/import_ftb/PackHelpers.cpp

```diff
@@ -74,17 +74,49 @@ Modpack parseDirectory(QString path)
         modpack.mcVersion = Json::requireString(root, "mcVersion", "mcVersion");
         modpack.jvmArgs = Json::ensureVariant(root, "jvmArgs", {}, "jvmArgs");
         modpack.totalPlayTime = Json::requireInteger(root, "totalPlayTime", "totalPlayTime");
+
+        auto modLoader = Json::ensureString(root, "modLoader", "modLoader");
+        if (!modLoader.isEmpty()) {
+            const auto parts = modLoader.split('-', Qt::KeepEmptyParts);
+            if (parts.size() >= 2) {
+                const auto loader = parts.first().toLower();
+                modpack.version = parts.at(1).trimmed();
+                if (loader == "neoforge") {
+                    modpack.loaderType = ModPlatform::NeoForge;
+                } else if (loader == "forge") {
+                    modpack.loaderType = ModPlatform::Forge;
+                } else if (loader == "fabric") {
+                    modpack.loaderType = ModPlatform::Fabric;
+                } else if (loader == "quilt") {
+                    modpack.loaderType = ModPlatform::Quilt;
+                }
+            }
+        } else {            
+            legacyInstanceParsing(path, &modpack.loaderType, &modpack.loaderVersion);
+        }
     } catch (const Exception& e) {
         qDebug() << "Couldn't load ftb instance json: " << e.cause();
         return {};
     }
+    
+    auto iconFile = QFileInfo(FS::PathCombine(path, "folder.jpg"));
+    if (iconFile.exists() && iconFile.isFile()) {
+        modpack.icon = QIcon(iconFile.absoluteFilePath());
+    } else {  // the logo is a file that the first bit denotes the image tipe followed by the actual image data
+        modpack.icon = loadFTBIcon(FS::PathCombine(path, ".ftbapp", "logo"));
+    }
+    return modpack;
+}
 
+void legacyInstanceParsing(QString path, std::optional<ModPlatform::ModLoaderType>* loaderType, QString* loaderVersion)
```

Comment:
Wouldn't it be better to just use references, instead of pointers? 
```suggestion
void legacyInstanceParsing(QString path, std::optional<ModPlatform::ModLoaderType>& loaderType, QString& loaderVersion)
```

Reply:
A mutable reference IMO isn't the best idea. Passing a value into a function you wouldn't necessarily expect it to be able to modify it.